expression/aggregation/base_func.go

```diff
@@ -204,6 +206,12 @@ func (a *baseFuncDesc) typeInfer4BitFuncs(ctx sessionctx.Context) {
 	// TODO: a.Args[0] = expression.WrapWithCastAsInt(ctx, a.Args[0])
 }
 
+func (a *baseFuncDesc) typeInfer4JsonFuncs(ctx sessionctx.Context) {
+	a.RetTp = types.NewFieldType(mysql.TypeJSON)
+	types.SetBinChsClnFlag(a.RetTp)
+	a.RetTp.Flag |= mysql.NotNullFlag
```

Comment:
according to mysql document [`JSON_OBJECTAGG(key, value)`](https://dev.mysql.com/doc/refman/5.7/en/group-by-functions.html#function_json-objectagg):
> Returns NULL if the result contains no rows, or in the event of an error

I think we should not add a not null flag on the result field type.